buildpacks/metrics-agent/src/layers/install_agentmon.rs

```diff
@@ -0,0 +1,329 @@
+
+    untar(agentmon_tgz.path(), destination)?;
+
+    chmod_plus_x(&destination.join("agentmon")).map_err(DownloadAgentmonError::PermissionError)?;
+
+    Ok(())
+}
+
+pub(crate) fn untar(
+    path: impl AsRef<Path>,
+    destination: impl AsRef<Path>,
+) -> Result<(), DownloadAgentmonError> {
+    let file =
+        fs_err::File::open(path.as_ref()).map_err(DownloadAgentmonError::CouldNotOpenFile)?;
+
+    Archive::new(GzDecoder::new(file))
+        .unpack(destination.as_ref())
+        .map_err(DownloadAgentmonError::CouldNotUnpack)
+}
+
+/// Sets file permissions on the given path to 7xx (similar to `chmod +x <path>`)
+pub fn chmod_plus_x(path: &Path) -> Result<(), std::io::Error> {
+    let mut perms = fs_err::metadata(path)?.permissions();
+    let mut mode = perms.mode();
+    octal_executable_permission(&mut mode);
+    perms.set_mode(mode);
+
+    fs_err::set_permissions(path, perms)
+}
+
+/// Write a script to the target path while adding a bash shebang line and setting execution permissions
+fn write_bash_script(path: &Path, script: impl AsRef<str>) -> std::io::Result<()> {
+    let script = script.as_ref();
+    fs_err::write(path, format!("#!/usr/bin/env bash\n\n{script}"))?;
+    chmod_plus_x(path)?;
+
+    Ok(())
+}
+
+/// Ensures the provided octal number's executable
+/// bit is enabled.
+///
+/// i.e. chmod +x will ensure that the first digit
+/// of the file permission is 7 on unix so if you pass
+/// in 0o455 it would be mutated to 0o755
+fn octal_executable_permission(mode: &mut u32) {
+    *mode |= 0o700;
+}
+
+pub(crate) fn download(
+    uri: impl AsRef<str>,
+    destination: impl AsRef<Path>,
+) -> Result<(), DownloadAgentmonError> {
+    let mut response_reader = ureq::get(uri.as_ref())
+        .call()
+        .map_err(|err| DownloadAgentmonError::RequestError(Box::new(err)))?
+        .into_reader();
+
+    let mut destination_file = fs_err::File::create(destination.as_ref())
+        .map_err(DownloadAgentmonError::CouldNotCreateDestinationFile)?;
+
+    std::io::copy(&mut response_reader, &mut destination_file)
+        .map_err(DownloadAgentmonError::CouldNotWriteDestinationFile)?;
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_chmod() {
+        let tmp = tempfile::tempdir().unwrap();
+        let file = tmp.into_path().join("file");
+        std::fs::write(&file, "lol").unwrap();
+
+        let before = file.metadata().unwrap().permissions().mode();
+        chmod_plus_x(&file).unwrap();
+        let after = file.metadata().unwrap().permissions().mode();
+        assert!(before != after);
+    }
+
+    #[test]
+    fn test_executable_logic() {
+        // Sets executable bit
+        let mut mode = 0o455;
+        octal_executable_permission(&mut mode);
+        assert_eq!(0o755, mode);
+
+        // Does not affect already executable
+        let mut mode = 0o745;
+        octal_executable_permission(&mut mode);
+        assert_eq!(0o745, mode);
+    }
```

Comment:
This essentially tests that bitwise OR works in Rust. I don't think we need that at all. There is only _one_ call-site for   `octal_executable_permission` which can easily replaced by inlining the `|= 0o700` making the function itself and the tests for it redundant.

Reply:
I moved this to a function specifically so I could test it as my original implementation was incorrect. Working with bitwise operators is something I've done for school, but rarely in Rails work. The value of the test, and docs are that future programmers have that info at their fingers without looking it up.